In [1]:
import pandas as pd

In [2]:
df_city = pd.read_csv("dim_cities.csv")

df_date = pd.read_csv("dim_date.csv")

df_plan= pd.read_csv("dim_plan.csv")

df_atliqo = pd.read_csv("fact_atliqo_metrics.csv")

df_mark_share = pd.read_csv("fact_market_share.csv")

df_revenue = pd.read_csv("fact_plan_revenue.csv")


df_metrics = pd.read_excel("metrics_list.xlsx")

In [3]:
df_city.head()

city_code  city_name
0     400001     Mumbai
1     110001      Delhi
2     700001    Kolkata
3     560001  Bangalore
4     600001    Chennai

In [4]:
df_date

date month_name before/after_5g  time_period
0  01-Jan-22        Jan       Before 5G            1
1  01-Feb-22        Feb       Before 5G            2
2  01-Mar-22        Mar       Before 5G            3
3  01-Apr-22        Apr       Before 5G            4
4  01-Jun-22        Jun        After 5G            1
5  01-Jul-22        Jul        After 5G            2
6  01-Aug-22        Aug        After 5G            3
7  01-Sep-22        Sep        After 5G            4

In [5]:
df_plan.head()

plan                                   plan_description
0   p1  Smart Recharge Pack (2 GB / Day Combo For 3 mo...
1   p2  Super Saviour Pack (1.5 GB / Day Combo For 56 ...
2   p3        Elite saver Pack (1 GB/ Day) Valid: 28 Days
3   p4  Mini Data Saver Pack (500 MB/ Day) Valid: 20 Days
4   p5                    Rs. 99 Full Talktime Combo Pack

In [6]:
df_atliqo

date  city_code company  atliqo_revenue_crores  arpu  \
0    01-Jan-22     400001  Atliqo                  60.69   192   
1    01-Jan-22     110001  Atliqo                  42.71   175   
2    01-Jan-22     700001  Atliqo                  36.91   175   
3    01-Jan-22     560001  Atliqo                  37.40   175   
4    01-Jan-22     600001  Atliqo                  30.80   203   
..         ...        ...     ...                    ...   ...   
115  01-Sep-22     800008  Atliqo                  11.86   250   
116  01-Sep-22     641001  Atliqo                   8.59   177   
117  01-Sep-22     160017  Atliqo                  10.32   163   
118  01-Sep-22     122001  Atliqo                   6.35   238   
119  01-Sep-22     492001  Atliqo                   3.73   173   

     active_users_lakhs  unsubscribed_users_lakhs  
0                 31.61                      1.90  
1                 24.41                      1.31  
2                 21.09                      1.25  
3                 21.37                      1.39  
4                 15.17                      1.12  
..                  ...                       ...  
115                4.74                      0.38  
116                4.85                      0.53  
117                6.33                      0.63  
118                2.67                      0.21  
119                2.16                      0.18  

[120 rows x 7 columns]

## PREPARE PLANS TABLE

In [65]:
df_revenue.head()

date  city_code plans  plan_revenue_crores
0  01-Jan-22     400001    p1                 6.26
1  01-Jan-22     110001    p1                 4.58
2  01-Jan-22     700001    p1                 4.29
3  01-Jan-22     560001    p1                 4.26
4  01-Jan-22     600001    p1                 3.60

There are 120 unique "date" and "city_code" pairs which is equal to the rows of df_atliqo.

**Merge with City**

In [9]:
df_rev_city = df_revenue.merge(df_city, how = 'left', on = 'city_code')

**Merge with date**

In [10]:

df_rev_city_date = df_rev_city.merge(df_date, how = 'left', on = 'date')

**Merge with plan descriptions**

In [11]:
df_rev_city_date_plan = df_rev_city_date.merge(df_plan, how = 'left', left_on = 'plans', right_on = 'plan')

In [12]:
# Dropping the plan column as it is redundant
df_rev_city_date_plan.drop(columns = 'plan', inplace = True)

In [78]:
df_rev_city_date_plan.head()

date  city_code plans  plan_revenue_crores  city_name month_name  \
0  01-Jan-22     400001    p1                 6.26     Mumbai        Jan   
1  01-Jan-22     110001    p1                 4.58      Delhi        Jan   
2  01-Jan-22     700001    p1                 4.29    Kolkata        Jan   
3  01-Jan-22     560001    p1                 4.26  Bangalore        Jan   
4  01-Jan-22     600001    p1                 3.60    Chennai        Jan   

  before/after_5g  time_period  \
0       Before 5G            1   
1       Before 5G            1   
2       Before 5G            1   
3       Before 5G            1   
4       Before 5G            1   

                                    plan_description  
0  Smart Recharge Pack (2 GB / Day Combo For 3 mo...  
1  Smart Recharge Pack (2 GB / Day Combo For 3 mo...  
2  Smart Recharge Pack (2 GB / Day Combo For 3 mo...  
3  Smart Recharge Pack (2 GB / Day Combo For 3 mo...  
4  Smart Recharge Pack (2 GB / Day Combo For 3 mo...

In [79]:
#df_rev_city_date_plan.to_csv("atliqo_plans.csv", index = False)

## Prepare Total Data

In [66]:
df_atliqo.head()

date  city_code company  atliqo_revenue_crores  arpu  \
0  01-Jan-22     400001  Atliqo                  60.69   192   
1  01-Jan-22     110001  Atliqo                  42.71   175   
2  01-Jan-22     700001  Atliqo                  36.91   175   
3  01-Jan-22     560001  Atliqo                  37.40   175   
4  01-Jan-22     600001  Atliqo                  30.80   203   

   active_users_lakhs  unsubscribed_users_lakhs  
0               31.61                      1.90  
1               24.41                      1.31  
2               21.09                      1.25  
3               21.37                      1.39  
4               15.17                      1.12

**Merge with City**

In [67]:
df_atliqo_city = df_atliqo.merge(df_city, how = 'left', on = 'city_code')  

**Merge with Date**

In [68]:

df_atliqo_city_date = df_atliqo_city.merge(df_date, how = 'left', on = 'date')

In [69]:
df_atliqo_city_date.head()

date  city_code company  atliqo_revenue_crores  arpu  \
0  01-Jan-22     400001  Atliqo                  60.69   192   
1  01-Jan-22     110001  Atliqo                  42.71   175   
2  01-Jan-22     700001  Atliqo                  36.91   175   
3  01-Jan-22     560001  Atliqo                  37.40   175   
4  01-Jan-22     600001  Atliqo                  30.80   203   

   active_users_lakhs  unsubscribed_users_lakhs  city_name month_name  \
0               31.61                      1.90     Mumbai        Jan   
1               24.41                      1.31      Delhi        Jan   
2               21.09                      1.25    Kolkata        Jan   
3               21.37                      1.39  Bangalore        Jan   
4               15.17                      1.12    Chennai        Jan   

  before/after_5g  time_period  
0       Before 5G            1  
1       Before 5G            1  
2       Before 5G            1  
3       Before 5G            1  
4       Before 5G            1

### Feature Engineering

In [70]:
df_atliqo_city_date["revenue_lost_crores"] = df_atliqo_city_date["unsubscribed_users_lakhs"]* df_atliqo_city_date["arpu"]/100

In [71]:
df_atliqo_city_date["perc_revenue_lost"] = (df_atliqo_city_date["revenue_lost_crores"] / df_atliqo_city_date["atliqo_revenue_crores"])*100

In [75]:
df_atliqo_city_date.tail()

date  city_code company  atliqo_revenue_crores  arpu  \
115  01-Sep-22     800008  Atliqo                  11.86   250   
116  01-Sep-22     641001  Atliqo                   8.59   177   
117  01-Sep-22     160017  Atliqo                  10.32   163   
118  01-Sep-22     122001  Atliqo                   6.35   238   
119  01-Sep-22     492001  Atliqo                   3.73   173   

     active_users_lakhs  unsubscribed_users_lakhs   city_name month_name  \
115                4.74                      0.38       Patna        Sep   
116                4.85                      0.53  Coimbatore        Sep   
117                6.33                      0.63  Chandigarh        Sep   
118                2.67                      0.21     Gurgaon        Sep   
119                2.16                      0.18      Raipur        Sep   

    before/after_5g  time_period  revenue_lost_crores  perc_revenue_lost  
115        After 5G            4               0.9500           8.010118  
116        After 5G            4               0.9381          10.920838  
117        After 5G            4               1.0269           9.950581  
118        After 5G            4               0.4998           7.870866  
119        After 5G            4               0.3114           8.348525

In [76]:
df_atliqo_city_date.to_csv('atliqo_total.csv', index=False) 

## Market Share Dataset

In [24]:
df_mark_share.head()

date  city_code  tmv_city_crores company  ms_pct
0  01-Jan-22     400001           286.29  Atliqo   21.20
1  01-Jan-22     110001           241.59  Atliqo   17.68
2  01-Jan-22     700001           222.19  Atliqo   16.61
3  01-Jan-22     560001           195.41  Atliqo   19.14
4  01-Jan-22     600001           166.78  Atliqo   18.47

In [25]:
df_mark_share_city =  df_mark_share.merge(df_city, how = 'left', on = ['city_code']) 

In [26]:
df_mark_share_city

date  city_code  tmv_city_crores company  ms_pct   city_name
0    01-Jan-22     400001           286.29  Atliqo   21.20      Mumbai
1    01-Jan-22     110001           241.59  Atliqo   17.68       Delhi
2    01-Jan-22     700001           222.19  Atliqo   16.61     Kolkata
3    01-Jan-22     560001           195.41  Atliqo   19.14   Bangalore
4    01-Jan-22     600001           166.78  Atliqo   18.47     Chennai
..         ...        ...              ...     ...     ...         ...
595  01-Sep-22     800008            68.74  Others    5.87       Patna
596  01-Sep-22     641001            60.90  Others    7.22  Coimbatore
597  01-Sep-22     160017            44.33  Others    7.35  Chandigarh
598  01-Sep-22     122001            34.48  Others    7.30     Gurgaon
599  01-Sep-22     492001            21.50  Others    7.83      Raipur

[600 rows x 6 columns]

In [27]:
df_mark_share_city_period = df_mark_share_city.merge(df_date, how = 'left', on = ['date']) 

In [28]:
df_mark_share_city_period

date  city_code  tmv_city_crores company  ms_pct   city_name  \
0    01-Jan-22     400001           286.29  Atliqo   21.20      Mumbai   
1    01-Jan-22     110001           241.59  Atliqo   17.68       Delhi   
2    01-Jan-22     700001           222.19  Atliqo   16.61     Kolkata   
3    01-Jan-22     560001           195.41  Atliqo   19.14   Bangalore   
4    01-Jan-22     600001           166.78  Atliqo   18.47     Chennai   
..         ...        ...              ...     ...     ...         ...   
595  01-Sep-22     800008            68.74  Others    5.87       Patna   
596  01-Sep-22     641001            60.90  Others    7.22  Coimbatore   
597  01-Sep-22     160017            44.33  Others    7.35  Chandigarh   
598  01-Sep-22     122001            34.48  Others    7.30     Gurgaon   
599  01-Sep-22     492001            21.50  Others    7.83      Raipur   

    month_name before/after_5g  time_period  
0          Jan       Before 5G            1  
1          Jan       Before 5G            1  
2          Jan       Before 5G            1  
3          Jan       Before 5G            1  
4          Jan       Before 5G            1  
..         ...             ...          ...  
595        Sep        After 5G            4  
596        Sep        After 5G            4  
597        Sep        After 5G            4  
598        Sep        After 5G            4  
599        Sep        After 5G            4  

[600 rows x 9 columns]

In [29]:
#df_mark_share_city_period.to_csv('atliqo_mrkt_share.csv', index=False) 

In [38]:
df_mark_share_city_period["Revenue_crores"] = (df_mark_share_city_period["ms_pct"]/100)*(df_mark_share_city_period["tmv_city_crores"])

In [40]:
df_mark_share_city_period.head()

date  city_code  tmv_city_crores company  ms_pct  city_name  \
0  01-Jan-22     400001           286.29  Atliqo   21.20     Mumbai   
1  01-Jan-22     110001           241.59  Atliqo   17.68      Delhi   
2  01-Jan-22     700001           222.19  Atliqo   16.61    Kolkata   
3  01-Jan-22     560001           195.41  Atliqo   19.14  Bangalore   
4  01-Jan-22     600001           166.78  Atliqo   18.47    Chennai   

  month_name before/after_5g  time_period  Revenue_crores  
0        Jan       Before 5G            1       60.693480  
1        Jan       Before 5G            1       42.713112  
2        Jan       Before 5G            1       36.905759  
3        Jan       Before 5G            1       37.401474  
4        Jan       Before 5G            1       30.804266

In [51]:
df_total_market = (df_mark_share_city_period.groupby(["before/after_5g","company"])["Revenue_crores"].sum()/df_mark_share_city_period.groupby("before/after_5g")["Revenue_crores"].sum()*100).reset_index()
df_total_market 


before/after_5g   company  Revenue_crores
0        After 5G    Atliqo       18.536479
1        After 5G    Britel       28.485480
2        After 5G  DADAFONE       10.357092
3        After 5G    Others        7.199483
4        After 5G       PIO       35.421467
5       Before 5G    Atliqo       19.892094
6       Before 5G    Britel       28.020790
7       Before 5G  DADAFONE       10.174544
8       Before 5G    Others        7.080093
9       Before 5G       PIO       34.832479

In [52]:
#df_total_market.to_csv('total_market_share.csv', index = False)

In [84]:
df_plan_contribution= (df_rev_city_date_plan.groupby(['before/after_5g','plans'])["plan_revenue_crores"].sum()/df_rev_city_date_plan.groupby('before/after_5g')["plan_revenue_crores"].sum()*100).reset_index()

df_plan_contribution

before/after_5g plans  plan_revenue_crores
0         After 5G    p1            22.338494
1         After 5G   p11            17.404856
2         After 5G   p12            10.869728
3         After 5G   p13             2.943709
4         After 5G    p2            13.921077
5         After 5G    p3            12.131451
6         After 5G    p4             8.206818
7         After 5G    p5             6.098018
8         After 5G    p6             4.629439
9         After 5G    p7             1.456411
10       Before 5G    p1            20.543536
11       Before 5G   p10             1.485771
12       Before 5G    p2            16.863674
13       Before 5G    p3            14.951778
14       Before 5G    p4            12.187631
15       Before 5G    p5            11.385247
16       Before 5G    p6             8.489636
17       Before 5G    p7             6.600406
18       Before 5G    p8             4.921972
19       Before 5G    p9             2.570350

In [85]:
df_plan_contribution.to_csv("df_plan_contribution.csv", index = False)

In [81]:
df_rev_city_date_plan.groupby('before/after_5g')["plan_revenue_crores"].sum()

before/after_5g
After 5G     1068.38
Before 5G     882.37
Name: plan_revenue_crores, dtype: float64

In [83]:
df_atliqo_city_date.groupby('before/after_5g')["atliqo_revenue_crores"].sum()

before/after_5g
After 5G     1589.66
Before 5G    1597.70
Name: atliqo_revenue_crores, dtype: float64